server/platform/shared/filestore/s3store.go

```diff
@@ -315,8 +334,14 @@ func (b *S3FileBackend) FileModTime(path string) (time.Time, error) {
 }
 
 func (b *S3FileBackend) CopyFile(oldPath, newPath string) error {
-	oldPath = b.prefixedPath(oldPath)
-	newPath = b.prefixedPath(newPath)
+	oldPath, err := b.prefixedPath(oldPath)
+	if err != nil {
+		return errors.Wrapf(err, "unable to prefix path %s", oldPath)
+	}
+	newPath, err = b.prefixedPath(newPath)
+	if err != nil {
+		return errors.Wrapf(err, "unable to prefix path %s", newPath)
+	}
```

Comment:
I stumbled over doing this for `newPath` (here and other writes below), given that I assume the intent is for all new files to use the proper prefixed path. Trying the unprefixed path for a new path seems fundamentally wrong, even if it's unlikely to actually exist.

Reply:
Fair point. Given that all this code will go away anyways, do you want me to handle this case? (Will probably need a flag or something to bypass the StatFile check)

Reply:
I fixed this btw. Realized that this will happen for the WriteFile case as well.